software/cooler/merge/main.nf

```diff
@@ -0,0 +1,40 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process COOLER_MERGE {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::cooler=0.8.11" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/cooler:0.8.11--pyh3252c3a_0"
+    } else {
+        container "quay.io/biocontainers/cooler:0.8.11--pyh3252c3a_0"
+    }
+
+    input:
+    tuple val(meta), path(cool)
+
+    output:
+    tuple val(meta), path("*.cool"), emit: cool
+    path "*.version.txt"           , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def cool_bin = "${meta.bin}"
```

Comment:
Why is `meta.bin` linked to a sample? I would have thought it would be something that is more configurable by the user? If so, then it needs to be provided as a non-mandatory argument vis `options.args`. Basically same comment as with `cooler/zoomify`.

Reply:
Yes this is the same as cooler/zoomify. I am not sure the best way to add the parameter to options.args. Maybe I can treat it as suffix.